packages/mana/basevector.go

```diff
@@ -133,3 +135,59 @@ func (bmv *BaseManaVector) GetWeightedMana(nodeID identity.ID, weight float64) (
 func (bmv *BaseManaVector) GetMana(nodeID identity.ID) (float64, error) {
 	return bmv.GetWeightedMana(nodeID, 0.5)
 }
+
+// ForEach iterates over the vector and calls the provided callback.
+func (bmv *BaseManaVector) ForEach(callback func(ID identity.ID, bm *BaseMana) bool) {
+	for nodeID, baseMana := range bmv.vector {
+		if !callback(nodeID, baseMana) {
+			return
+		}
+	}
+}
+
+//GetManaMap return mana perception of the node.
+func (bmv *BaseManaVector) GetManaMap() Map {
+	res := make(map[identity.ID]float64)
+	for ID := range bmv.vector {
+		mana, _ := bmv.GetMana(ID)
+		res[ID] = mana
+	}
+	return res
+}
+
+// GetHighestManaNodes return the n highest mana nodes in descending order.
+// It also updates the mana values for each node.
+func (bmv *BaseManaVector) GetHighestManaNodes(n int) []NodeIDMana {
```

Comment:
either make `n` a `uint` or throw an error if it is negative

Reply:
Going with `uint`